### ISSUE TITLE
Incorrect example code for `Lint/UselessAccessModifier`

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12652]
https://github.com/rubocop/rubocop/blob/b4080d8efc42466f31c63a443157363b11a467f7/lib/rubocop/cop/lint/useless_access_modifier.rb#L114-L126

This says that the Rails method `delegate` respects access modifiers, however that doesn't seem to be the case:

```rb
require "active_support/core_ext/module/delegation"
require "ostruct"
class Bar
  def qux
    OpenStruct.new(baz: "bar")
  end
end

class Foo < Bar
  private
  delegate :baz, to: :qux # private: true is what you'd want
end

puts Foo.new.baz # works just fine
```

This is further supported by `delegate` having a keyword argument called `private` to achieve the desired behaviour: https://api.rubyonrails.org/classes/Module.html#method-i-delegate

I would change docs myself but can't actually think of a usecase for this config value.

**Comment by vlad-pisanov** [comment]
The `delegate` example is definitely wrong.

`MethodCreatingMethods` is for things like `define_method` that are aware of their current visibility. A somewhat contrived example would be:

```ruby
class C
  private # wrongly flagged as Lint/UselessAccessModifier: Useless private access modifier
  
  send :define_method, :foo, -> { 'E621' } # `private` is respected when `foo` is defined
end

C.new.foo
# private method `foo' called for #<C:0x00007fa1770405b8> (NoMethodError)
```
